### PR TITLE
Add a separate grafana.com API URL setting

### PR DESCRIPTION
**Pull request by sd2k**
The `GrafanaComURL` setting is currently used in two places:

- the `/api/gnet` endpoint, which proxies all requests to the URL
  configured in `GrafanaComURL`
- OAuth logins using grafana.com, where the auth URL, token URL and
  redirect URL are all configured to use the `GrafanaComURL`.

This has worked fine until now because almost all Grafana instances have
just used the default value, https://grafana.com. However, we now have a
few different grafana.com's, some of which are behind IAP. The IAP
causes the `/api/gnet` proxy to fail because the required cookies are not
present in the request (how could they be?). Setting the
`[grafana_net.url]` setting to an internal-only URL improves the situation
slightly - the proxy works again just fine - but breaks any OAuth logins
using grafana.com, because the user must be redirected to a publicly
accessible URL.

This commit adds an additional setting, `[grafana_net.api_url]`
(and equivalently `[grafana_com.api_url]`), which can be used to tell
Grafana to use the new API URL when proxying requests to the grafana.com
API, while still using the existing `GrafanaComURL` setting for other
things.

The setting will fall back to the `GrafanaComURL` setting + "/api" if unset.

### CODE REVIEW

File: pkg/setting/setting.go
Comment (sakjur) [inline pkg/setting/setting.go:181]: QQ: Is it possible for us to avoid this global variable?

File: conf/defaults.ini
Comment (sakjur) [inline conf/defaults.ini:1060]: WDYT about skipping the `[grafana_net.api_url]` and just providing `[grafana_com.api_url]`?

File: pkg/setting/setting.go
Comment (sd2k) [reply pkg/setting/setting.go:181]: Yeah I think so, wasn't sure if they were still around for a reason so I just left them in. I'll get rid of it 👍

File: conf/defaults.ini
Comment (sd2k) [reply conf/defaults.ini:1060]: Sounds good to me!

File: conf/defaults.ini
Comment (papagian) [reply conf/defaults.ini:1064]: I was about to suggest the same but on 2nd thought are these settings meaningful for other users except `grafana_com` so that they need to override them?

File: pkg/setting/setting.go
Comment (sakjur) [reply pkg/setting/setting.go:181]: Ah, just legacy, the `Cfg` struct is preferred 🙂

File: conf/defaults.ini
Comment (sakjur) [reply conf/defaults.ini:1064]: 👍 Yeah, that's a fair point. I'll leave it up to you @sd2k.

Comment (sd2k) [reply conf/defaults.ini]: Hm I added it anyway. It might come in handy for docs internally 🤷

**Review by sakjur — APPROVED** [review]
👍 lgtm!

**Comment by sd2k** [comment]
@sakjur thanks! I'm not sure exactly what to do about the milestone / backport - I don't think this needs backporting, FWIW.

**Comment by sakjur** [comment]
@sd2k I added the `no-backport` label and set the milestone to the next minor release for you, that should do it 👍 
Checked the enterprise tests, and the failures seem unrelated, so good to merge afaict

**Comment by sd2k** [comment]
Thanks @sakjur, I don't have merge rights on grafana/grafana sadly, could you merge for me? thanks!

**Comment by sakjur** [comment]
@sd2k ✔️🎉

**Pull request merged**
